ocs_sample_library_preview/BaseClient.py

```diff
@@ -133,4 +133,4 @@ def checkResponse(self, response, main_message: str):
     def request(self, method: str, url: str, params=None, data=None, headers=None, **kwargs):
         if not headers:
             headers = self.sdsHeaders()
-        return requests.request(method, url, params=params, data=data, headers=headers, **kwargs)
+        return requests.request(method, requests.utils.quote(url, safe=':/'), params=params, data=data, headers=headers, **kwargs)
```

Comment:
In case you are wondering why this is here, the requests library was not properly URL encoding # symbols, which the SDS endpoint does not like. Therefore, I added the requests.utils.quote function to url encode the url ahead of time, while leaving ```:``` and ```/``` characters.